src/frontegg-auth/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use base64::{decode_config, URL_SAFE};
+use derivative::Derivative;
+use jsonwebtoken::{decode, Algorithm, DecodingKey, Validation};
+use uuid::Uuid;
+
+fn try_base64_decode<T: AsRef<[u8]>>(input: T) -> Vec<u8> {
+    let input = input.as_ref();
+    decode_config(input, URL_SAFE).unwrap_or_else(|_| input.to_vec())
+}
+
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
+pub struct FronteggAuthentication {
+    pub admin_api_token_url: String,
+    #[derivative(Debug = "ignore")]
+    pub decoding_key: DecodingKey,
+    pub tenant_id: Uuid,
+}
+
+impl FronteggAuthentication {
+    /// Creates a new frontegg auth. `jwk_rsa_pem` is the RSA public key to
+    /// validate the JWTs and can optionally be base64 encoded (with the URL-safe
+    /// alphabet). `tenant_id` must be parseable as a UUID.
+    pub fn new(
+        admin_api_token_url: String,
+        jwk_rsa_pem: &[u8],
+        tenant_id: &str,
+    ) -> Result<Self, anyhow::Error> {
+        // Attempt to base64 decode the password, otherwise keep it as is. This is
+        // unambiguous because the URL-safe base64 alphabet does not contain newline,
+        // which is present in the key.
+        let jwk_rsa_pem = try_base64_decode(&jwk_rsa_pem);
```

Comment:
If you are trying to pass this as a cli flag you need a way to remove the dashes and newlines, and base64 is much easier than...whatever you have to do in bash to do it correctly (I never figured it out). Couldn't use env vars at the time because I was deploying this over mzcloud with its extra args thing. We can possibly just remove the base64 thing from this because cloud is going to do it all in env vars and we won't be doing adhoc deployments for testing.

Reply:
I've got this wired up in cloud to pass the PEM as an environment variable verbatim, it is fairly easy - don't think we need many provisions for cmdline args for our specific use case.